File: rewrite-extensions.py
import os
import re

directory = 'src'

for root, dirs, files in os.walk(directory):
    for filename in files:
        if filename.endswith(('.tsx', '.ts', '.json')):
            filepath = os.path.join(root, filename)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()

            new_content = re.sub(r'\.(mp4|mov)', '.webm', content, flags=re.IGNORECASE)
            new_content = re.sub(r'\.(png|jpg|jpeg)', '.webp', new_content, flags=re.IGNORECASE)

            if content != new_content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated: {filepath}")

File: rewrite-extensions.js
const fs = require('fs');
const path = require('path');

const directoryPath = path.join(__dirname, 'src');

function walkDir(dir) {
    fs.readdirSync(dir).forEach(f => {
        let dirPath = path.join(dir, f);
        let isDirectory = fs.statSync(dirPath).isDirectory();
        if (isDirectory) {
            walkDir(dirPath);
        } else {
            if (dirPath.endsWith('.tsx') || dirPath.endsWith('.ts') || dirPath.endsWith('.json')) {
                let content = fs.readFileSync(dirPath, 'utf8');
                
                // Replace videos
                let newContent = content.replace(/\.mp4|\.mov/gi, '.webm');
                
                // Replace images
                newContent = newContent.replace(/\.png|\.jpg|\.jpeg/gi, '.webp');
                
                if (content !== newContent) {
                    fs.writeFileSync(dirPath, newContent, 'utf8');
                    console.log(`Updated formatting in: ${dirPath}`);
                }
            }
        }
    });
}

console.log('Starting massive media extension rewrite...');
walkDir(directoryPath);
console.log('Extensions universally updated to web fallback equivalents.');
